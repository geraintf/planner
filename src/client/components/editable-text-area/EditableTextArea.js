import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './EditableTextArea.less';

export default class EditableTextArea extends Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
    onBlur: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    onKeyPress: PropTypes.func.isRequired
  };

  constructor(props) {
    super(props);

    this.textArea = React.createRef();

  }

  componentDidMount() {
    this.textArea.current.focus();
  }

  render() {
    const {
      text,
      onBlur,
      onChange,
      onKeyPress
    } = this.props;

    return (
      <div className="editable-text-area">
        <textarea
          ref={this.textArea}
          value={text}
          onBlur={onBlur}
          onChange={onChange}
          onKeyPress={onKeyPress}
        />
      </div>
    );
  }
}
