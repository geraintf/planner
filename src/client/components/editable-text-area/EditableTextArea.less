.editable-text-area {
  textarea {
    height: 55px;
    resize: none;
    width: 100%;
  }
}