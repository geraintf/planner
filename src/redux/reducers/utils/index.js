import uuidv4 from 'uuid/v4';
import { arrayMove } from 'react-sortable-hoc';

import { getDayTodos } from '../../selectors';

import { syncDayTodos } from '../../../services/requests';

export const toggleTodos = (state, todoId) => {
  const toggle = (todos,id) => todos.map(todo => (
    todo.todoId === id
      ? { ...todo, completed: !todo.completed }
      : todo
  ));

  const updatedTodos = {
    ...state.todos,
    [state.selectedDate]: toggle(getDayTodos(state), todoId)
  };

  syncDayTodos(state.selectedDate, updatedTodos[state.selectedDate]);

  return {
    ...state,
    todos: updatedTodos
  };
};

export const addTodo = (state, text) => {
  const newTodo = {
    todoId: uuidv4(),
    text,
    completed: false
  };

  const updatedTodos = {
    ...state.todos,
    [state.selectedDate]: [...getDayTodos(state)].concat([newTodo])
  };

  syncDayTodos(state.selectedDate, updatedTodos[state.selectedDate]);

  return {
    ...state,
    todos: updatedTodos
  };
};


export const removeTodo = (state, todoId) => {
  const updatedTodos = {
    ...state.todos,
    [state.selectedDate]: getDayTodos(state).filter(todo => todo.todoId !== todoId)
  };

  syncDayTodos(state.selectedDate, updatedTodos[state.selectedDate]);

  return {
    ...state,
    todos: updatedTodos
  };
};

export const moveTodo = (state, oldIndex, newIndex) => {
  const updatedTodos = {
    ...state.todos,
    [state.selectedDate]: arrayMove(getDayTodos(state), oldIndex, newIndex)
  };

  syncDayTodos(state.selectedDate, updatedTodos[state.selectedDate]);

  return {
    ...state,
    todos: updatedTodos
  };
};

export const editTodo = (state, todoId, newValue) => {
  const edit = (todos, id, newValue) =>
    todos.map(todo => {
      if (todo.todoId === id) todo.text = newValue;

      return todo;
    });

  const updatedTodos = {
    ...state.todos,
    [state.selectedDate]: edit(getDayTodos(state), todoId, newValue)
  };

  syncDayTodos(state.selectedDate, updatedTodos[state.selectedDate]);

  return {
    ...state,
    todos: updatedTodos
  };
};
