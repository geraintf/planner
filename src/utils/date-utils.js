import { format } from 'date-fns';

const DATE_FORMAT = 'DD-MM-YYYY';

const formatTodoDateKey = date => format(date, DATE_FORMAT).toString();

export {
  formatTodoDateKey
};
