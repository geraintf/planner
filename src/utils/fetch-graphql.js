require('isomorphic-fetch');

const opts = {
  method: 'POST',
  headers: { 'Content-Type': 'application/json' },
  credentials: 'same-origin'
};

export const makeGraphqlReq = ({ query, operationName, variables }, path = '') => {

  if(!query) {
    throw new Error('An invalid query was passed into makeGraphqlReq: ' + query);
  }

  return fetch(`${path}/graphql`, {
    ...opts,
    body: JSON.stringify({ query, operationName, variables })
  }).then(res => res.json());
};
