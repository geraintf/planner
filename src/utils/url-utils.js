
export const getFullUrl = req => req.protocol + '://' + req.get('host');
