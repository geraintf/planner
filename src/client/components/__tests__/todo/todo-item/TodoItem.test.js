import React from 'react';
import  { shallow } from 'enzyme';

import { unwrapComponent } from '../../../../../../test/utils/unwrap';

import WrappedTodoItem from '../../../todo/todo-item/TodoItem';
import EditableTextArea from '../../../editable-text-area/EditableTextArea';

const TodoItem = unwrapComponent(WrappedTodoItem);

describe('TodoItem', () => {
  let output;
  let props;

  beforeEach(() => {
    props = {
      editTodo: () => {},
      onClick: () => {},
      completed: false,
      text: 'oh hey didnt see you there',
      index: 0,
      id: '123'

    };
    output = shallow(<TodoItem {...props} />);
  });

  it('should render correctly', () => {
    expect(output.dive()).toMatchSnapshot();
  });

  describe('given the todo item is being edited', () => {
    it('should render just the EditableTextArea ', () => {
      output.setState({ editing: true });

      const todoField = output.dive().find('TodoField').dive();

      expect(todoField.type()).toEqual(EditableTextArea);
    });
  });

  describe('given the todo item is not being edited', () => {
    it('should render just the todo text ', () => {
      output.setState({ editing: false });

      const todoField = output.dive().find('TodoField').dive();

      expect(todoField.text()).toEqual('oh hey didnt see you there');
    });
  });
});
