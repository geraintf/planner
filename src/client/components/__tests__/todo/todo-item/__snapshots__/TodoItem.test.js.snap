// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should render correctly 1`] = `
<li
  className="todo-item"
>
  <button
    onClick={[Function]}
  />
  <span
    className="todo-item__content"
    onDoubleClick={[Function]}
  >
    <TodoField />
  </span>
  <div
    className="todo-item__utils"
  >
    <Connect(TodoUtilsMenu)
      todoId="123"
      toggleEditing={[Function]}
    />
    <Component />
  </div>
</li>
`;
